internal/exec/stages/disks/partitions.go

```diff
@@ -353,7 +354,23 @@ func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 		case exists && shouldExist && matches:
 			s.Logger.Info("partition %d found with correct specifications", part.Number)
 		case exists && shouldExist && !wipeEntry && !matches:
-			return fmt.Errorf("Partition %d didn't match: %v", part.Number, matchErr)
+			number := part.Number == info.Number
```

Comment:
I'd suggest moving these tests into a function so it's easy for the "didn't match" error to specifically explain what didn't match.  The structure I'd use is:

```
// existing function
func partitionMatches(...)
// new function for this case
func partitionMatchesResize(...)
// helper function called by both of the above functions, which handles the common tests
func partitionMatchesCommon(...)
```